pkg/operators/combiner/combiner.go

```diff
@@ -171,11 +184,17 @@ func (o *combinerOperatorInstance) forwardData(
 	var c <-chan time.Time
 
 	if config.interval == 0 {
-		// Define a maximum waiting time for data from all targets
-		// TODO: Make it configurable?
-		timeout := time.NewTimer(5 * time.Second)
-		defer timeout.Stop()
-		c = timeout.C
+		if config.count == 0 {
+			// We are only handling array data sources here
+			// If interval and count are both 0, wait till the user stops the gadget
+			c = nil
+		} else {
+			// Define a maximum waiting time for data from all targets
+			// TODO: Make it configurable?
+			timeout := time.NewTimer(5 * time.Second)
+			defer timeout.Stop()
+			c = timeout.C
+		}
```

Comment:
The first conditional isn't needed. (even if I agree the comments are easier to write with both branches)

```suggestion
		// If count is 0, wait until the user stops the gadget, otherwise wait for
		// 5 seconds
		// TODO: Make it configurable?
		if config.count != 0 {
			timeout := time.NewTimer(5 * time.Second)
			c = timeout.C
			defer timeout.Stop()
		}
```